minio/helpers.py

```diff
@@ -156,11 +200,11 @@ def is_valid_bucket_name(bucket_name):
     :return: True if the bucket is valid. Raise :exc:`InvalidBucketError` otherwise.
     """
     ## verify bucket name length.
-    if len(bucket_name) < 3: 
+    if len(bucket_name) < 3:
         raise InvalidBucketError('Bucket name cannot be less than 3 characters.')
     if len(bucket_name) > 63:
         raise InvalidBucketError('Bucket name cannot be more than 63 characters.')
-    
+
     validbucket = re.compile('^[a-zA-Z][a-zA-Z0-9\\-]+[a-zA-Z0-9]$')
```

Comment:
Please compile regex at only once.  Here it will be compiled every function call.